### PR TITLE
Added C622 - Locations --> Temporary Location --> (Validate matching …

### PR DESCRIPTION
**Pull request by KazachenkaTatsiana**
…settings)

https://foliotest.testrail.io/index.php?/cases/view/622
https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5649/console
image

### CODE REVIEW

File: cypress/support/fragments/inventory/holdingsRecordView.js [hunk 1]
```diff
@@ -227,4 +227,5 @@ export default {
     cy.do(Button(including('Record last updated:')).click());
     cy.expect(HTML(including(`Source: ${userName}`)).exists());
   },
+  verifyRecordViewingOpen: () => cy.expect(HTML({ id: 'paneHeaderui-inventory.holdingsRecordView' }).exists()),
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/holdingsRecordView.js:230]: What this method for?
Just use waitLoading()

File: cypress/support/fragments/inventory/holdingsRecordEdit.js [hunk 2]
```diff
@@ -163,4 +165,12 @@ export default {
   chooseCallNumberType(type) {
     cy.do(callNumberType.choose(type));
   },
+  verifyEditViewingOpen: () => cy.expect(HTML({ id: 'clickable-create-holdings-record' }).exists()),
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/holdingsRecordEdit.js:168]: What this method for?
Just use waitLoading()

File: cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        );
+      });
+
+      cy.visit(settingsMenu.tenantLocationsPath);
+      Locations.waitLoading();
+      [...Array(3)].forEach((_, index) => {
+        Locations.selectInstitution(itemData.instances[index].defaultLocation.institutionName);
+        Locations.selectCampus(itemData.instances[index].defaultLocation.campusName);
+        Locations.selectLibrary(itemData.instances[index].defaultLocation.libraryName);
+        Locations.openLocationDetails(itemData.instances[index].defaultLocation.name);
```
Comment (manowarqqq) [inline cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js:131]: There is no expected result for Step3

File: cypress/support/fragments/inventory/holdingsRecordEdit.js [hunk 4]
```diff
@@ -32,6 +32,8 @@ const statisticalCodeFieldSet = FieldSet('Statistical code');
 const addStatisticalCodeButton = Button('Add statistical code');
 const callNumberType = rootForm.find(Select('Call number type'));
 const statisticalCodeSelectionList = statisticalCodeFieldSet.find(SelectionList());
+const additemTemporarylocation = Button({ id: 'additem_temporarylocation' });
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/holdingsRecordEdit.js:35]: temporaryLocationDropdown

File: cypress/support/fragments/inventory/holdingsRecordEdit.js [hunk 5]
```diff
@@ -163,4 +165,12 @@ export default {
   chooseCallNumberType(type) {
     cy.do(callNumberType.choose(type));
   },
+  verifyEditViewingOpen: () => cy.expect(HTML({ id: 'clickable-create-holdings-record' }).exists()),
+  openTemporarylocation() {
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/holdingsRecordEdit.js:169]: openTemporaryLocation

File: cypress/support/fragments/inventory/holdingsRecordEdit.js [hunk 6]
```diff
@@ -163,4 +165,12 @@ export default {
...
+    cy.do(additemTemporarylocation.click());
+  },
+  verifyTemporaryLocationItemExists: (temporarylocation) => {
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/holdingsRecordEdit.js:172]: temporaryLocationList.find(SelectionOption(temporarylocation)).exists()

File: cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+  after('Delete  test data', () => {
+    cy.getAdminToken();
+    InventoryInstances.deleteInstanceAndHoldingRecordAndAllItemsViaApi(itemData.barcode);
+    [...Array(3)].forEach((_, index) => {
+      Location.deleteViaApiIncludingInstitutionCampusLibrary(
+        itemData.instances[index].defaultLocation.institutionId,
+        itemData.instances[index].defaultLocation.campusId,
+        itemData.instances[index].defaultLocation.libraryId,
+        itemData.instances[index].defaultLocation.id,
+      );
```
Comment (manowarqqq) [inline cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js:102]: Delete Location for Inventory inst
Delete Service Point

File: cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+import getRandomPostfix from '../../../support/utils/stringTools';
+import Location from '../../../support/fragments/settings/tenant/locations/newLocation';
+import { ITEM_STATUS_NAMES } from '../../../support/constants';
+import InventorySearchAndFilter from '../../../support/fragments/inventory/inventorySearchAndFilter';
+import HoldingsRecordView from '../../../support/fragments/inventory/holdingsRecordView';
+import HoldingsRecordEdit from '../../../support/fragments/inventory/holdingsRecordEdit';
+import { Locations } from '../../../support/fragments/settings/tenant';
+import settingsMenu from '../../../support/fragments/settingsMenu';
+
+describe('Temporary Location', () => {
```
Comment (TetianaParanich) [inline cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js:14]: please use two describes like
describe('inventory', () => {
describe('Item', () => {
and put all tests to 'e2e\inventory\item' folder

File: cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+import Location from '../../../support/fragments/settings/tenant/locations/newLocation';
+import { ITEM_STATUS_NAMES } from '../../../support/constants';
+import InventorySearchAndFilter from '../../../support/fragments/inventory/inventorySearchAndFilter';
+import HoldingsRecordView from '../../../support/fragments/inventory/holdingsRecordView';
+import HoldingsRecordEdit from '../../../support/fragments/inventory/holdingsRecordEdit';
+import { Locations } from '../../../support/fragments/settings/tenant';
+import settingsMenu from '../../../support/fragments/settingsMenu';
+
+describe('inventory', () => {
+describe('item', () => {
```
Comment (TetianaParanich) [inline cypress/e2e/inventory/holdings/locations-temporary-validate-matching-settings.cy.js:15]: please fix linter issue

**Review by TetianaParanich — APPROVED** [review]

**Pull request merged**